nifi-extension-bundles/nifi-couchbase-bundle/nifi-couchbase-services-api/src/main/java/org/apache/nifi/services/couchbase/exception/CouchbaseErrorHandler.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.services.couchbase.exception;
+
+import java.util.Map;
+
+import static org.apache.nifi.services.couchbase.exception.CouchbaseErrorHandler.ErrorHandlingStrategy.Failure;
+
+public class CouchbaseErrorHandler {
+
+    private final Map<Class<? extends Exception>, ErrorHandlingStrategy> exceptionMapping;
+
+    public CouchbaseErrorHandler(Map<Class<? extends Exception>, ErrorHandlingStrategy> exceptionMapping) {
+        this.exceptionMapping = exceptionMapping;
+    }
+
+    public ErrorHandlingStrategy getStrategy(Exception exception) {
+        return exceptionMapping.getOrDefault(exception.getClass(), Failure);
+    }
+
+    public enum ErrorHandlingStrategy {
+        Rollback, Failure, Retry
```

Comment:
In general, `enum`  values should be uppercased. Is there a reason for this naming convention?